### ISSUE TITLE
Running role second time corrupts config.toml

### ISSUE THREAD

**Issue by tschoonj** [riemers/ansible-gitlab-runner#155]
After upgrading this role from v1.6.18 to v1.6.23 I observed that the role is no longer idempotent as I get this failure when running my ansible script for the second time:

```
TASK [riemers.gitlab-runner : (Windows) Verify config] *******************************************************************************************************************************************
fatal: [vm-ip]: FAILED! => {"changed": true, "cmd": "c:/gitlab-runner//gitlab-runner.exe verify", "delta": "0:00:00.082744", "end": "2020-12-12 08:00:30.054828", "msg": "non-zero return code", "rc": 1, "start": "2020-12-12 08:00:29.972084", "stderr": "Runtime platform                                  \u001b[0;m  arch\u001b[0;m=amd64 os\u001b[0;m=windows pid\u001b[0;m=4444 revision\u001b[0;m=8fa89735 version\u001b[0;m=13.6.0\n\u001b[31;1mFATAL: Near line 30 (last key parsed 'runners.docker.allowed_images'): Key 'runners.docker.allowed_images' has already been defined.\u001b[0;m \n", "stderr_lines": ["Runtime platform                                  \u001b[0;m  arch\u001b[0;m=amd64 os\u001b[0;m=windows pid\u001b[0;m=4444 revision\u001b[0;m=8fa89735 version\u001b[0;m=13.6.0", "\u001b[31;1mFATAL: Near line 30 (last key parsed 'runners.docker.allowed_images'): Key 'runners.docker.allowed_images' has already been defined.\u001b[0;m "], "stdout": "", "stdout_lines": []}
```

Here is a screenshot of the corrupted config.toml file:

<img width="1742" alt="Screenshot 2020-12-12 at 08 08 34" src="https://user-images.githubusercontent.com/65736/101979185-c3821880-3c52-11eb-8155-645e46dd21ab.png">

**Comment by scatmanoleg** [comment]
The same issue:
```
TASK [riemers.gitlab-runner-v1.6.23 : Assemble new config.toml] ********************************************************************************************************************************************************************
fatal: [hz-runner4]: FAILED! => {"changed": false, "msg": "failed to validate: rc:1 error:Runtime platform                                  \u001b[0;m  arch\u001b[0;m=amd64 os\u001b[0;m=linux pid\u001b[0;m=29484 revision\u001b[0;m=8fa89735 version\u001b[0;m=13.6.0\nRunning in system-mode.                           \u001b[0;m \n                                                  \u001b[0;m \n\u001b[31;1mFATAL: Near line 22 (last key parsed 'runners.cache.Shared'): Key 'runners.cache.Shared' has already been defined.\u001b[0;m \n"}
```

**Comment by riemers** [comment]
If someone can find it, it was introduced in one of the releases, but at the moment i just don't have the time. I can merge PR's but thats about it.

**Comment by scatmanoleg** [comment]
I've made some investigation (and will continue) but maybe someone can help me.
The problem became in this issue (and PR): https://github.com/riemers/ansible-gitlab-runner/issues/153

-------------

```
My config:
[[runners]]
  name = "runner1"
  limit = 0
  output_limit = 4096
  url = ""
  token = ""
  executor = "docker"
  environment = ["DOCKER_TLS_CERTDIR=/certs"]
  [runners.custom_build_dir]
  [runners.cache]
    "Shared" = true
    "Type" = "s3"
    "Path" = "cache"
    [runners.cache.azure]
  [runners.docker]
    "cpus" = "4"
    "volumes" = ["/certs/client", "/cache"]
    "privileged" = true
    "memory" = "8g"
    image = "alpine"
    disable_entrypoint_overwrite = false
    oom_kill_disable = false
    disable_cache = false
    shm_size = 0
    [runners.cache.s3]
      "ServerAddress" = "s3.amazonaws.com"
      "Insecure" = false
      "BucketLocation" = "eu-west-1"
      "BucketName" = "runners-cache"
      "SecretKey" = ""
      "AccessKey" = ""
```

-------------

```
- name: Modify existing line {{ line }}
  lineinfile:
    path: "{{ temp_runner_config.path }}"
    insertafter: '\s+\[{{ section | regex_escape }}\]'
    regexp: '^(\s*){{ line| regex_escape }} ='
    line: '{{ "  " * (section.split(".")|length) }}{{ line | to_json }} = {{ gitlab_runner.extra_configs[section][line] | to_json }}'
  register: modified_config_line
```
We have added `to_json` to the "line" parameter and parameters stopped to match regexp:
Before:
```
  [runners.docker]
    cpus = "4"
    volumes = ["/certs/client", "/cache"]
    privileged = true
    memory = "8g"
```
After:
```
  [runners.docker]
    "cpus" = "4"
    "volumes" = ["/certs/client", "/cache"]
    "privileged" = true
    "memory" = "8g"
```
I've tried to fix it with adding `to_json` to the "regexp" parameters and it works:
`regexp: '^(\s*){{ line| to_json | regex_escape }} ='`
And this is fixing error from that issue.

Unfortunately if delete some parameters (for example `cpus: '4'`) it doesn't remove from final toml. I will try to figure out why it happens, for sure I need to check some other regexp-s.

---------------------
But I realised that there is another issue, at the second and next running sorting breaks:
```
    shm_size = 0
      "ServerAddress" = "s3.amazonaws.com"
      "Insecure" = false
      "BucketLocation" = "eu-west-1"
      "BucketName" = "runners-cache"
      "SecretKey" = ""
      "AccessKey" = ""
    [runners.cache.s3]
```
I've checked it in a version before the fix with to_json and it has similar behaviour.
```
    disable_cache = false
    shm_size = 0
      BucketName = "runners-cache"
    [runners.cache.s3]
      ServerAddress = "s3.amazonaws.com"
      Insecure = false
      BucketLocation = "eu-west-1"
      SecretKey = ""
      AccessKey = ""
```

Looks like (from my point of view) it is better to fix both issues together, cause if I fix only the first, role will run without issues, but we will have corrupted configs.

**Comment by scatmanoleg** [comment]
https://github.com/riemers/ansible-gitlab-runner/pull/157

**Comment by maglub** [comment]
Hi, since you merged #157 you can close this issue.
